
# HG changeset patch
# User Darren Salt <linux@youmustbejoking.demon.co.uk>
# Date 1203549555 0
# Node ID 0a9d58e399a14cfadc622f562020ba5123254592
# Parent 6f70d1625834502a735349a6597ad7a85fb68aa5
Off-by-one in the FLAC security fix, breaking playback of some files.

--- a/src/demuxers/demux_flac.c	Wed Feb 20 21:45:48 2008 +0000
+++ b/src/demuxers/demux_flac.c	Wed Feb 20 23:19:15 2008 +0000
@@ -202,7 +202,7 @@ static int open_flac_file(demux_flac_t *
 
           length = _X_LE_32(ptr);
           ptr += 4 + length;
-          if (length >= block_length - 8)
+          if (length > block_length - 8)
             return 0; /* bad length or too little left in the buffer */
 
           user_comment_list_length = _X_LE_32(ptr);

